pkg/table/column.go

```diff
@@ -624,8 +624,13 @@ func getColDefaultValue(ctx expression.BuildContext, col *model.ColumnInfo, defa
 }
 
 func getColDefaultValueFromNil(ctx expression.BuildContext, col *model.ColumnInfo, args *getColOriginDefaultValue) (types.Datum, error) {
-	if !mysql.HasNotNullFlag(col.GetFlag()) && !mysql.HasNoDefaultValueFlag(col.GetFlag()) {
-		return types.Datum{}, nil
+	if !mysql.HasNotNullFlag(col.GetFlag()) {
+		if !mysql.HasNoDefaultValueFlag(col.GetFlag()) ||
+			(col.GetDefaultValue() == nil && col.GetOriginDefaultValue() == nil) {
+			// In CanSkip function(in table/tables pkg), if column's default value is nil, and the column value is NULL too,
+			// then the column value can be skipped and won't encode to row value.
```

Comment:
Maybe we need to check HasNoDefaultValueFlag in the `CanSkip` function too?

But this fix is still needed, otherwise, the same write in the old version will also have this problem.

Reply:
Seems this fix should only affect `GetColOriginDefaultValue` (nil is from `OringinDefaultValue`), can we change the bool to `isOriginalDefaultValueNil` ?